test/box-luatest/gh_6421_mvcc_wrong_index_count_test.lua

```diff
@@ -0,0 +1,60 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all = function()
+    g.server = server:new{
+        alias   = 'default',
+        box_cfg = {memtx_use_mvcc_engine = true}
+    }
+    g.server:start()
+end
+
+g.after_all = function()
+    g.server:drop()
+end
+
+g.test_mvcc_wrong_space_count = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local fiber = require('fiber')
+
+        local s = box.schema.space.create('test')
+        s:create_index('primary')
+
+        local f1 = fiber.create(function()
+            fiber.self():set_joinable(true)
+            s:insert{1}
+        end)
+
```

Comment:
Ditto: please use `ERRINJ_WAL_DELAY`.

Reply:
If we use ERRINJ_WAL_DELAY, we cannot test release build. On the other hand we can rely on strict order of evaluation here.